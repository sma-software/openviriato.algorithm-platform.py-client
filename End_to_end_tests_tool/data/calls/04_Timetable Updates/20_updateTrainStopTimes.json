{
    "restResource": "/trains/{trainID}/train-path-nodes:update-trajectory-stop-times",
    "httpMethod": "PUT",
    "urlParameters": [ "=shortTrain:ID=" ],
    "requestBody": {
        "TrainPathNodeID": "=shortTrain:TrainPathNodes[0].ID=",
        "ArrivalTime": "2003-05-01T00:04:00",
        "DepartureTime": "2003-05-01T00:05:30",
        "StopStatus": "operationalStop"
    },
    "CSharpMethodName": "UpdateTrajectory(trainID, updateStopTimesTrainPathNode)",
    "CSharpMethodParameterTypes": [ "System.Int64", "SMA.AlgorithmPlatform.AlgorithmInterface.AIDM.IUpdateStopTimesTrainPathNode" ],
    "subcategory": "Update Train Stop Times"
}
