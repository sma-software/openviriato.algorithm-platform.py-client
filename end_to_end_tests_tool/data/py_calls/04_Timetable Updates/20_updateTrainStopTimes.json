{
  "PythonParameterFromJson" :
  [
      {
          "ParameterName" : "train_id",
          "aidm_factory": "str",
          "ObjectJson":
          {
              "object": "2060"
          }
      },
      {
          "ParameterName": "update_train_times_node",
          "aidm_factory": "convert_json_to_update_train_stop_times_node",
          "ObjectJson":
          {
              "TrainPathNodeID": "1332",
              "ArrivalTime": "2003-05-01T00:04:00",
              "DepartureTime": "2003-05-01T00:05:30",
              "StopStatus": "operational_stop"
          }
      }
  ]
}
