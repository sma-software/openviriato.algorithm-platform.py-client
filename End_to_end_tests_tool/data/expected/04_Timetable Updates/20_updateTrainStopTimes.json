{
  "ID":2060,
  "TrainPathNodes":
  [
    {
      "ID":1332,
      "SectionTrackID":null,
      "NodeID":18,
      "NodeTrackID":null,
      "FormationID":1187,
      "ArrivalTime":"2003-05-01T00:04:00",
      "DepartureTime":"2003-05-01T00:05:30",
      "MinimumRunTime":null,
      "MinimumStopTime":"P0D",
      "StopStatus":"OperationalStop",
      "SequenceNumber":0
    },
    {
      "ID":1696,
      "SectionTrackID":1171,
      "NodeID":10,
      "NodeTrackID":null,
      "FormationID":null,
      "ArrivalTime":"2003-05-01T00:10:30",
      "DepartureTime":"2003-05-01T00:10:30",
      "MinimumRunTime":"PT5M",
      "MinimumStopTime":"P0D",
      "StopStatus":"CommercialStop",
      "SequenceNumber":1
    }
  ],
  "DebugString":"RVZH_1_1_J03 tt_(G)"
}